simvue/run.py

```diff
@@ -121,6 +124,37 @@ def validate_timestamp(timestamp):
 
     return True
 
+def get_filename_input(input_data, name_tag):
+    """ 
+    Determines if the input is pickleable or a valid file. If it is pickleable and not a string then a pickle file is created 
+    and filename is returned.
+
+    If the input is a string, so a filename, then check if the file exists, else raise and exception
+
+    Parameters
+    ----------
+    input_data : pickleable object or file
+       
+    name_tag: A tag to be added to the filename. Typically the run name.
+
+    Returns
+    -------
+    filename
+        The name of the file either pointing to the pickleable object or file
+    """          
+    filename=input_data
+    if dill.pickles(input_data) and isinstance(input_data, str) == False:
+        filename=name_tag +'_'+randomname.get_name() + '.pkl'
+        filehandle = open(filename, 'wb')
```

Comment:
Why write to a temporary file? It would be more straightforward to just directly send the contents of the pickled object to the object storage.

Reply:
This was to cater to offline mode, without having to check for that and also for users to have a local copy. This may be unnecessary but some users may prefer it. The other issue is that if there are lots of files then having a local store will be an issue, so perhaps go with a direct store and add it later if needed.